### DISCUSSION TITLE
The new TeslaOne requirements

### DISCUSSION THREAD

**Discussion by snarl817** [jasonacox/pypowerwall#121] [General]
I noticed that my dashboard hadn't had any data since the end of October. The pypowerwall logs were full of "invalid login" errors. When I logged in to the gateway from a web browser, it told me I needed to switch to TeslaOne.

Now, that's the LAST thing I want to do. I had an electrician install a Universal Wall Connector so I could charge my PHEV, and I discovered that I had NO ability to login to the internal server for provisioning or configuration...I HAD to use TeslaOne. The issue I had was  that when I tried to add the wall connector to my "project", it wanted stuff like pictures of the inside of the panel and the connector showing the wiring. Luckily, the max my car pulls is 16A, so I have no need to tell the wall connector anything. It's on a 60A breaker, so it can safely pull 48A. Still, people on Reddit are talking about TeslaOne like it's the greatest thing since sliced bread.

I WAS able to get the password reset and the pypowerwall container updated, and now I have data flowing into influx, but I'm concerned: If Tesla discontinues local API access in favor of TeslaOne, will the dashboard support it? I have yet to find a step-by-step guide on adding the gateway to TeslaOne, and frankly, I'm worried that if I do so, it'll disable local API access. This dashboard is the ONLY reason that if I buy a new house, I'd be insisting on solar/powerwall from Tesla.

**Comment by jasonacox** [comment]
> This dashboard is the ONLY reason that if I buy a new house, I'd be insisting on solar/powerwall from Tesla.

❤️ ❤️ ❤️ ❤️  Thanks @snarl817 - made my day. 😁 

> When I logged in to the gateway from a web browser, it told me I needed to switch to TeslaOne.

You should still be able to bypass that by pushing "Continue" and then "Upgrade Later" on those screens: https://github.com/jasonacox/Powerwall-Dashboard/discussions/109#discussioncomment-7278795

> If Tesla discontinues local API access in favor of TeslaOne

It uses the TEDAPI access point which is how we get extended metrics and you can switch to only using that via `setup.sh` and selecting "Powerwall 3" mode.  See https://github.com/jasonacox/Powerwall-Dashboard?tab=readme-ov-file#extended-device-vitals-metrics-beta.
